src/Integration.Vsix/CFamily/CFamilyReproducerCommand.cs

```diff
@@ -58,8 +57,8 @@ public static async Task InitializeAsync(AsyncPackage package, ILogger logger)
             await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(package.DisposalToken);
 
             var monitorSelection = await package.GetServiceAsync(typeof(SVsShellMonitorSelection)) as IVsMonitorSelection;
-            var adapterService = await package.GetMefServiceAsync<IVsEditorAdaptersFactoryService>();
-            var docLocator = new ActiveDocumentLocator(monitorSelection, adapterService);
+            var textDocumentProvider = await package.GetMefServiceAsync<ITextDocumentProvider>();
+            var docLocator = new ActiveDocumentLocator(monitorSelection, textDocumentProvider);
```

Comment:
This way of construction doesn't seem quite right. If we are going to make one of our classes MEF-constructable, it should be `ActiveDocumentLocator`, since `ITextDocumentProvider` is an implementation detail as far the `CFamilyReproducerCommand` is concerned.

It might make more sense when `ITextDocumentProvider` is being used elsewhere but it looks odd here.

Reply:
Take a look at [WIP ](https://github.com/SonarSource/sonarlint-visualstudio/blob/4749d102e3ff5f20c10b61aeabb414503caeac2a/src/Integration.Vsix/Helpers/DocumentEvents/DocumentFocusedEventRaiser.cs#L66) of how it will be used